isisd/isis_tlvs.c

```diff
@@ -1970,6 +2477,76 @@ static int pack_items_(uint16_t mtid, enum isis_tlv_context context,
 		       struct list *new_fragment_arg);
 #define pack_items(...) pack_items_(ISIS_MT_IPV4_UNICAST, __VA_ARGS__)
 
+/* Functions related to Sub-Sub-TLVs in general */
+
+struct isis_subsubtlvs *isis_alloc_subsubtlvs(enum isis_tlv_context context)
+{
+	struct isis_subsubtlvs *result;
+
+	result = XCALLOC(MTYPE_ISIS_SUBSUBTLV, sizeof(*result));
```

Comment:
There are 2 functions for END.X : `srv6_endx_sid_add` and `srv6_endx_sid_del`.

Here `isis_alloc_subsubtlvs` is called from `srv6_endx_sid_add`. I see here the memory allocation when END.X sid is added.

We also want to have XFREE of this allocated memory in `srv6_endx_sid_del` right?

Reply:
we probably want to run this through some sort of memory leak check before committing, if possible ... lots of allocs and frees, we might won't catch all of them by inspection (?)

Reply:
@riw777 Thanks for the suggestion. I used valgrind to discover possible memory leaks and I fixed all of them.